pkg/downloader/manager.go

```diff
@@ -82,6 +82,19 @@ func (m *Manager) Build() error {
 
 	// Check that all of the repos we're dependent on actually exist.
 	req := c.Metadata.Dependencies
+
+	// If using apiVersion v1, calculate the hash before resolve repo names
+	// because resolveRepoNames will change req if req uses repo alias
+	// and Helm 2 calculate the digest from the original req
+	// Fix for: https://github.com/helm/helm/issues/7619
+	var v2Sum string
+	if c.Metadata.APIVersion == chart.APIVersionV1 {
+		v2Sum, err = resolver.HashV2Req(req)
+		if err != nil {
+			return err
```

Comment:
The `err` here is just error of calculating hash, not checking with the digest in lock file. 
Checking with Helm v2 digest will only happen when checking with v3 failed and the APIVersion is v1, and that error message you mentioned will be returned there.  
This is an issue caused by the different sequence of dependency update process between Helm v2 and v3. In v3, digest is calculated after resolving repo names. But in v2, it is before resolving repo names, and if repo is referred by alias, it will get changed after resolving name. So, to correctly check with v2 hash, we need to calculate the hash here.

Reply:
I understand you point about the error is not to do with the digest. However, sending a more user friendly message back to the user they can progress better. If you look at the code it also checked for not nil and returned this message.

Reply:
You are right. The error message is changed. I am sorry for so many time reviews.